From 9ae6d6e540a77a6cc8408c8c289f7915e2ec741e Mon Sep 17 00:00:00 2001
From: Grace Cham <hscham@chromium.org>
Date: Fri, 20 Jan 2023 13:47:40 +0900
Subject: [PATCH] Add forwarding headers
 base/{bind,callback{,_forward,_helpers}}.h

The headers were moved to base/functional/ in crrev.com/1048084. The old
ones were kept as temporary forwarding headers but now removed in
crrev.com/1093705 after migration was done on upstream.

BUG=b:

Change-Id: Ibc46d37d3c120c62cbc75a38c577e8fe6c582fa7
---
 base/bind.h             | 12 ++++++++++++
 base/callback.h         | 12 ++++++++++++
 base/callback_forward.h | 12 ++++++++++++
 base/callback_helpers.h | 12 ++++++++++++
 4 files changed, 48 insertions(+)
 create mode 100644 base/bind.h
 create mode 100644 base/callback.h
 create mode 100644 base/callback_forward.h
 create mode 100644 base/callback_helpers.h

diff --git a/base/bind.h b/base/bind.h
new file mode 100644
index 000000000000..721f61a9952f
--- /dev/null
+++ b/base/bind.h
@@ -0,0 +1,12 @@
+// Copyright 2022 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+//
+// TODO(https://crbug.com/1364441): Temporary forwarding header.
+
+#ifndef BASE_BIND_H_
+#define BASE_BIND_H_
+
+#include "base/functional/bind.h"
+
+#endif  // BASE_BIND_H_
diff --git a/base/callback.h b/base/callback.h
new file mode 100644
index 000000000000..d561a3df298b
--- /dev/null
+++ b/base/callback.h
@@ -0,0 +1,12 @@
+// Copyright 2022 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+//
+// TODO(https://crbug.com/1364441): Temporary forwarding header.
+
+#ifndef BASE_CALLBACK_H_
+#define BASE_CALLBACK_H_
+
+#include "base/functional/callback.h"
+
+#endif  // BASE_CALLBACK_H_
diff --git a/base/callback_forward.h b/base/callback_forward.h
new file mode 100644
index 000000000000..5d18a2541c8b
--- /dev/null
+++ b/base/callback_forward.h
@@ -0,0 +1,12 @@
+// Copyright 2022 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+//
+// TODO(https://crbug.com/1364441): Temporary forwarding header.
+
+#ifndef BASE_CALLBACK_FORWARD_H_
+#define BASE_CALLBACK_FORWARD_H_
+
+#include "base/functional/callback_forward.h"
+
+#endif  // BASE_CALLBACK_FORWARD_H_
diff --git a/base/callback_helpers.h b/base/callback_helpers.h
new file mode 100644
index 000000000000..324877d929af
--- /dev/null
+++ b/base/callback_helpers.h
@@ -0,0 +1,12 @@
+// Copyright 2022 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+//
+// TODO(https://crbug.com/1364441): Temporary forwarding header.
+
+#ifndef BASE_CALLBACK_HELPERS_H_
+#define BASE_CALLBACK_HELPERS_H_
+
+#include "base/functional/callback_helpers.h"
+
+#endif  // BASE_CALLBACK_HELPERS_H_
-- 
2.39.1.405.gd4c25cc71f-goog

